### ISSUE TITLE
Test failure: System.Security.Cryptography.X509Certificates.Tests.ChainTests.VerifyWithRevocation

### ISSUE THREAD

**Issue by v-haren** [dotnet/runtime#41952]
failed in job: [runtime-libraries-coreclr outerloop 20200907.3 ](https://dev.azure.com/dnceng/public/_build/results?buildId=804427&view=ms.vss-test-web.build-test-results-tab&runId=25478140&resultId=103383&paneView=debug)

net5.0-Linux-Release-arm64-CoreCLR_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-helix-arm64v8-bfcd90a-20200127194925

Error message
~~~
Assert.Equal() Failure
Expected: 3
Actual:   2


Stack trace
   at System.Security.Cryptography.X509Certificates.Tests.ChainTests.VerifyWithRevocation() in /_/src/libraries/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs:line 785
~~~

**Comment by vcsjones** [comment]
This is probably related to the expiring intermediate as noted in #40708. I’ll see if I can either fix this test or re-write it in terms of DynamicChainTests.

**Comment by vcsjones** [comment]
Actually I think this might have just been a networking hiccup or another intermittent issue. This test is passing for me locally. I tested in an Ubuntu 18.04 ARM64 environment and it still passes.

> dotnet test --filter VerifyWithRevocation /p:OuterLoop=true
>
> Test Run Successful.
Total tests: 1
     Passed: 1
 Total time: 2.8526 Seconds

**Comment by krwq** [comment]
@v-haren can you confirm this is a single instance or do you get consistent repro?

**Comment by v-haren** [comment]
I couldn't repro this. It did not appear in the recent failed.

**Comment by krwq** [comment]
Considering this looks like could have been caused by the network hiccup and we have only a single instance of this I'll close this issue as no repro. Please re-open if this happens again and we'll need to find some way to write a test in more reliable way.

**Comment by krwq** [comment]
Thanks @vcsjones for the investigation!

**Comment by v-haren** [comment]
failed again in job: [ runtime-libraries-coreclr outerloop 20200922.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=825084&view=ms.vss-test-web.build-test-results-tab&runId=26285674&resultId=103106&paneView=debug)

failed test: System.Security.Cryptography.X509Certificates.Tests.ChainTests.VerifyWithRevocation

net5.0-Linux-Release-arm64-CoreCLR_release-(Ubuntu.1804.ArmArch.Open)Ubuntu.1804.ArmArch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-helix-arm64v8-bfcd90a-20200127194925


Error message
~~~
Assert.Equal() Failure
Expected: 3
Actual:   2


Stack trace
   at System.Security.Cryptography.X509Certificates.Tests.ChainTests.VerifyWithRevocation() in /_/src/libraries/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs:line 785
~~~

**Comment by bartonjs** [comment]
I started having a long writeup as to why there might be an actual problem here that we're missing, because obviously `valid` was `true`.  Then I remembered that for some reason (Linux: because Windows does, Windows: ???) AllowUnknownCertificateAuthority means that X509Chain.Build returns true for PartialChain.

So, yeah, network timeout/flakiness on a system that doesn't already have the root for that chain cached.
